api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/IncomingConnectorFactory.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2018-2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.reactive.messaging.spi;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.MessagingProvider;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+
+import javax.enterprise.inject.spi.DeploymentException;
+import java.util.NoSuchElementException;
+
+/**
+ * SPI used to implement a connector managing a source of messages for a specific <em>transport</em>. For example, to
+ * handle the consumption of records from Kafka, the reactive messaging extension would need to implement a {@code bean}
+ * implementing this interface. This bean is called for every {@code channel} that needs to be created for this specific
+ * <em>transport</em> (so Kafka in this case). These channels are connected to methods annotated with
+ * {@link org.eclipse.microprofile.reactive.messaging.Incoming}.
+ * <p>
+ * The factory is called to create a {@code channel} for each configured <em>transport</em>. The configuration is done using
+ * MicroProfile Config. The following snippet gives an example for a hypothetical Kafka connector:
+ *
+ * <pre>
+ * mp.messaging.incoming.my-channel.type=i.e.m.reactive.messaging.impl.kafka.Kafka
```

Comment:
I think this attribute name should be "provider". This is because it allows us to comply
with the generic pattern for annotation parameter override of: 

mp.messaging.incoming|outgoing.channel-name.<parameter>=value

and the parameter in both annotations is called "provider". So we have
```
mp.messaging.incoming.my-channel.provider=value
```